src/OrleansAzureUtils/Storage/RemindersTableManager.cs

```diff
@@ -118,55 +118,57 @@ private RemindersTableManager(Guid serviceId, string deploymentId, string storag
-                        && String.Compare(e.PartitionKey, sBegin) > 0
-                        && String.Compare(e.PartitionKey, sEnd) <= 0;
-
-                var queryResults = await ReadTableEntriesAndEtagsAsync(query);
+                string filterBetweenBeginAndEnd = TableQuery.CombineFilters(
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThan, sBegin),
+                    TableOperators.And,
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual,
+                        sEnd));
+                string query = TableQuery.CombineFilters(filterOnServiceIdStr, TableOperators.And, filterBetweenBeginAndEnd);
+                var queryResults = await ReadTableEntriesAndEtagsWithFilterStringAsync(query);
                 return queryResults.ToList();
             }
 
             if (begin == end)
             {
-                
-                Expression<Func<ReminderTableEntry, bool>> query =
-                    e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                        && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0;
-
-                var queryResults = await ReadTableEntriesAndEtagsAsync(query);
+                var queryResults = await ReadTableEntriesAndEtagsWithFilterStringAsync(filterOnServiceIdStr);
                 return queryResults.ToList();
             }
 
             // (begin > end)
-            Expression<Func<ReminderTableEntry, bool>> p1Query =
-                e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                    && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0
-                    && String.Compare(e.PartitionKey, sBegin) > 0;
-            Expression<Func<ReminderTableEntry, bool>> p2Query =
-                e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                    && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0
-                    && String.Compare(e.PartitionKey, sEnd) <= 0;
-
-            var p1 = ReadTableEntriesAndEtagsAsync(p1Query);
-            var p2 = ReadTableEntriesAndEtagsAsync(p2Query);
-
-            IEnumerable<Tuple<ReminderTableEntry, string>>[] arr = await Task.WhenAll(p1, p2);
-            return  arr[0].Concat(arr[1]).ToList();
+            string filterExtendBeginAndEnd = TableQuery.CombineFilters(
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThan, sBegin),
+                    TableOperators.Or,
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual,
+                        sEnd));
+            string query1 = TableQuery.CombineFilters(
+                filterOnServiceIdStr, TableOperators.And, filterExtendBeginAndEnd);
+
+            var results = await ReadTableEntriesAndEtagsWithFilterStringAsync(query1);
+            return results.ToList();
```

Comment:
I understand these are logically equivalent, but I'm pretty sure they are not equivalent in performance. The new approach will probably force Azure to do a full table scan, while the other will do 2 faster partial scans, as well as parallelizing them. Since there is no need to have a single query to do both (even if the new approach would be just twice as slower than the parallel calls, which is probably some more than that), I'd stick to the original approach. If you feel strongly about it, we should have the change in a separate isolated PR anyway, so that it's not lost in the sea of style/new APIs usage fixes.